dev/NavigationView/NavigationView.cpp

```diff
@@ -524,29 +596,50 @@ void NavigationView::UpdateItemsRepeaterItemsSource(const winrt::ItemsRepeater&
 
 void NavigationView::OnNavigationViewItemIsSelectedPropertyChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
 {
-    if (auto const nvib = sender.try_as<winrt::NavigationViewItemBase>())
+    if (auto const nvi = sender.try_as<winrt::NavigationViewItem>())
     {
         // Check whether the container that triggered this call back is the selected container
-        bool isContainerSelectedInModel = IsContainerTheSelectedItemInTheSelectionModel(nvib);
-        bool isSelectedInContainer = nvib.IsSelected();
+        bool isContainerSelectedInModel = IsContainerTheSelectedItemInTheSelectionModel(nvi);
+        bool isSelectedInContainer = nvi.IsSelected();
 
         if (isSelectedInContainer && !isContainerSelectedInModel)
         {
-            auto indexPath = GetIndexPathForContainer(nvib);
-            m_selectionModel.SelectAt(indexPath);
+            auto indexPath = GetIndexPathForContainer(nvi);
+            UpdateSelectionModelSelection(indexPath);
```

Comment:
>UpdateSelectionModelSelection [](start = 12, length = 29)

Are seperator elements reflected in the selection model?

Reply:
Yep, the SelectionModel is aware of all items in the MenuItems/MenuItemsSource

---
In reply to: [386538413](https://github.com/microsoft/microsoft-ui-xaml/pull/2004#discussion_r386538413) [](ancestors = 386538413)